File: 0002-add-two-numbers/0002-add-two-numbers.js

function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

/**
* @param {ListNode} l1
* @param {ListNode} l2
* @return {ListNode}
*/
var addTwoNumbers = function (l1, l2) {
    let result = new ListNode();
    let seek = result;
    let addition = 0;
    while (l1 || l2) {
        console.log(l1.val, l2.val)
        let sum = l1.val + l2.val + addition;
        addition = Math.floor(sum / 10);
        sum = sum - addition*10;
        seek.val = sum;
        if(!l1.next && !l2.next){
            if(addition){
                seek.val = sum;
                seek.next = new ListNode(addition);
            } 
             break;

        }

        seek.next = new ListNode();
        seek = seek.next;
        l1 = l1.next ? l1.next : new ListNode();
        l2 = l2.next ? l2.next : new ListNode();
   }
   return result
};

File: 0002-add-two-numbers/0002-add-two-numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
	sum := 0
	var result ListNode
	var temp *ListNode
	addition := 0
	temp = &result
	for  {
		sum = 0

		if l1 != nil {
			sum += l1.Val
			l1 = l1.Next
		}
		if l2 != nil {
			sum += l2.Val
			l2 = l2.Next
		}
		
		sum += addition
		temp.Val = sum%10
		addition = sum / 10
		if l2 == nil && l1 == nil && addition == 0 {
			break
		}
		temp.Next = &ListNode{}
		temp = temp.Next
		

	}
	return &result
}